src/cortex_cli/cortex_cli.py

```diff
@@ -238,13 +239,55 @@ def status(config_file, verbose) -> None:
     time_left_rt = str(datetime.timedelta(seconds=seconds_rt))
     click.echo(f'Time left on refresh token (hh:mm:ss): {time_left_rt}')
 
-    active_pid = check_daemon(tokens_file)
+    active_pid = check_token_manager(tokens_file)
     if active_pid:
         click.echo(f'Token manager: {click.style("RUNNING", fg="green")} (PID {active_pid})')
     else:
         click.echo(f'Token manager: {click.style("NOT RUNNING", fg="red")}')
 
 
+def _validate_cortex_cli_login(config_file, no_daemon, no_refresh) -> dict:
+    """Checks if provided combination of auth login optionsis valid:
+       - no_daemon and no_refresh are mutually exclusive
+       - daemon mode should not be requested on Windows
+       - config file must exist
+
+    Args:
+        config_file (str): --config-file option value
+        no_daemon (bool): --no-daemon option value
+        no_refresh (bool): --no-refresh option value
+    Raises:
+        click.BadOptionUsage: if both mutually exclusive --no-daemon and --no-refresh are set
+        click.UsageError: if daemon is requested on Windows
+        click.BadParameter: if provided config_file does not exist or is invalid
+    Returns:
+        dict: config dict loaded from config_file
+    """
+
+    # --no-refresh and --no-daemon are mutually exclusive
+    if no_refresh and no_daemon:
+        raise click.BadOptionUsage(
+            '--no-refresh',
+            "Cannot request a non-daemonic (foreground) token manager when using '--no-refresh'.")
```

Comment:
Doesn't `click` provide functionality to do this automatically?

Reply:
Not that I know of. Did not find built in validators for combinations of options.
The closes thing is to attach a [validator callback](https://click.palletsprojects.com/en/8.1.x/options/?highlight=validate#callbacks-for-validation), but that works for validating the value of a single option. Here we need to validate the pair. 

I assume @kukushechkin also did not find a better way, as his [previous PR](https://github.com/iqm-finland/cortex-cli/pull/19) implements mutual exclusivity this way.

Reply:
Yes, indeed. One of the options might be another lib — `cloup` (it is a set of bells and whistles around `click`), it has constraints logic and can hide mutual exclusivity of options, but in my case "optional default value" for file paths wasn't possible and bringing another dependency instead of few lines of code didn't make much sense.

Reply:
I see, nvm. I had higher expectations from `click` ))